core/src/main/scala/akka/kafka/internal/ConsumerProgressTracking.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <https://softwaremill.com>
+ * Copyright (C) 2016 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.kafka.internal
+import org.apache.kafka.clients.consumer.{Consumer, OffsetAndMetadata}
+import org.apache.kafka.common.TopicPartition
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * Track the current state of the consumer: what offsets its has requested and committed, filtering by the current
+ * assignments to the consumer. When a partition is assigned the consumer for the first time, its assigned offset
+ * is the current position of the partition (uses underlying Kafka Consumer to leverage the configured offset-reset
+ * policy).
+ */
+trait ConsumerProgressTracking {
+  var committedOffsets: Map[TopicPartition, OffsetAndMetadata] = _
+  var requestedOffsets: Map[TopicPartition, OffsetAndMetadata] = _
```

Comment:
Should these be `def`s and only the implementation have access to change the values?